File: compiler-group-traslator/src/test/resources/c/BooleanNot.c
#include<stdio.h>
#include<stdlib.h>
#include<stdbool.h>
#include "GC.c"
#include "ExceptionHandlingFunctions.c"
VM* vm;
bool not(bool b) {
	void* RU = !b;
	return RU;

	//set all vars to zero and have the TEMP variable for the return expression
}
int main(char* args) {
	vm = newVM();
	assertTrue(!false == true);
	assertTrue(!true == false);
	assertTrue(not(false) == true);
	assertTrue(not(true) == false);
	return 0;
}

File: compiler-group-traslator/src/test/resources/BooleanNot.java
public class BooleanNot {
    public static void main(String[] args) {
        StaticJavaLib.assertTrue(!false == true);
        StaticJavaLib.assertTrue(!true  == false);

        StaticJavaLib.assertTrue(not(false) == true);
        StaticJavaLib.assertTrue(not(true)  == false);
    }

    static boolean not(boolean b) {
        return !b;
    }
}

